packages/cli/src/GenericHelpers.ts

```diff
@@ -214,4 +229,81 @@ export async function validateEntity(
 	}
 }
 
+/**
+ * Create an error execution
+ *
+ * @param {INode} node
+ * @param {IWorkflowDb} workflowData
+ * @param {Workflow} workflow
+ * @param {WorkflowExecuteMode} mode
+ * @returns
+ * @memberof ActiveWorkflowRunner
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export async function createErrorExecution(
+	error: ExecutionError,
+	node: INode,
+	workflowData: IWorkflowDb,
+	workflow: Workflow,
+	mode: WorkflowExecuteMode,
+) {
+	const executionData: IRunExecutionData = {
+		startData: {
+			destinationNode: node.name,
+			runNodeFilter: [node.name],
+		},
+		executionData: {
+			contextData: {},
+			nodeExecutionStack: [
+				{
+					node,
+					data: {
+						main: [
+							[
+								{
+									json: {},
+									pairedItem: {
+										item: 0,
+									},
+								},
+							],
+						],
+					},
+					source: null,
+				},
+			],
+			waitingExecution: {},
+			waitingExecutionSource: {},
+		},
+		resultData: {
+			runData: {
+				[node.name]: [
+					{
+						startTime: 0,
+						executionTime: 0,
+						error,
+						source: [],
+					},
+				],
+			},
+			error,
+			lastNodeExecuted: node.name,
+		},
+	};
+
+	const fullExecutionData: IExecutionDb = {
+		data: executionData,
+		mode,
+		finished: false,
+		startedAt: new Date(),
+		workflowData,
+		workflowId: workflow.id,
+		stoppedAt: new Date(),
+	};
+
+	const execution = ResponseHelper.flattenExecutionData(fullExecutionData);
+
```

Comment:
@krynble Should we check the workflow settings to check whether save error execution is enabled?

<img width="1248" alt="image" src="https://user-images.githubusercontent.com/16496553/188723760-f8d75d10-b52c-40ac-9510-fc5ca87d454a.png">

Reply:
Hum that's a good point. Now thinking about alternatives, we have what we call "Activation errors".

You can see in the `packages/cli/src/ActiveWorkflowRunner.ts` file specifically inside `add` function, we try to activate a workflow and if it fails, we deactivate it and add an error the user can see when opening the workflows list.

Instead of generating a failed execution, what if we deactivate the workflow and display this error?

On the other hand, this is bad because we'll deactivate the workflow even if the service is temporarily down and a next poll request would work just fine.

Maybe the current approach you implemented is best for debuggability and failed executions are saved by default, so we should check this flag, yes, and only save if it's checked.

We just need to make sure that `WorkflowData` that we received was actually populated with this, because I've seen situations where this did not happen.

Reply:
Yeah, I think this is a much better approach. For example, in the Airtable case, the issue is generated by changing the trigger column in Airtable (Workflow does not have to be touched). In case this happens, with this approach, the user will have an issue in the error tab or will be notified via the error workflow, then look at the error, and finally adjust in Airtable without having to touch the workflow. The other approach will be an extra step having to activate the workflow again, and if we do not save the execution plus they do not have an error workflow setup, they will never notice the error.